helix-core/src/test/java/org/apache/helix/manager/zk/TestZkHelixAdmin.java

```diff
@@ -560,6 +560,82 @@ public void testLegacyEnableDisablePartition() {
+    String clusterName = className + "_" + methodName;
+    String instanceName = "TestInstance";
+    String testResource = "TestResource";
+    System.out.println("START " + clusterName + " at " + new Date(System.currentTimeMillis()));
+    HelixAdmin admin = new ZKHelixAdmin(_gZkClient);
+    admin.addCluster(clusterName, true);
+    admin.addInstance(clusterName, new InstanceConfig(instanceName));
+    admin.enableInstance(clusterName, instanceName, true);
+
+    InstanceConfig instanceConfig = admin.getInstanceConfig(clusterName, instanceName);
+
+    // Test the sanity check for resetPartition.
+    // resetPartition is expected to throw an exception when provided with a nonexist instance.
+    try {
+      admin.resetPartition(clusterName, "WrongTestInstance", testResource, Arrays.asList("1", "2"));
+      Assert.fail("Should throw HelixException");
+    } catch (HelixException expected) {
+      // This exception is expected because the instance name is made up.
+      Assert.assertEquals(expected.getMessage(), String.format(
+          "Can't reset state for %s.[1, 2] on WrongTestInstance, because WrongTestInstance"
+              + " has never existed in cluster %s", testResource, clusterName));
+    }
+
+    // resetPartition is expected to throw an exception when provided with a non-live instance.
+    try {
+      admin.resetPartition(clusterName, instanceName, testResource, Arrays.asList("1", "2"));
+      Assert.fail("Should throw HelixException");
+    } catch (HelixException expected) {
+      // This exception is expected because the instance is not alive.
+      Assert.assertEquals(expected.getMessage(), String.format(
+          "Can't reset state for %s.[1, 2] on %s, because %s does not live in cluster %s anymore",
+          testResource, instanceName, instanceName, clusterName));
+    }
+
+    HelixManager manager = initializeHelixManager(clusterName, instanceConfig.getInstanceName());
+    manager.connect();
+
+    // resetPartition is expected to throw an exception when provided with a non-exist resource.
+    try {
+      admin.resetPartition(clusterName, instanceName, testResource, Arrays.asList("1", "2"));
+      Assert.fail("Should throw HelixException");
+    } catch (HelixException expected) {
+      // This exception is expected because the resource is not added.
+      Assert.assertEquals(expected.getMessage(), String
+          .format("Can't reset state for %s.[1, 2] on %s, because %s is not added", testResource,
+              instanceName, testResource));
+    }
+
+    IdealState idealState = new IdealState(testResource);
+    idealState.setNumPartitions(3);
+    admin.addStateModelDef(clusterName, "MasterSlave", new MasterSlaveSMD());
+    idealState.setStateModelDefRef("MasterSlave");
+    idealState.setRebalanceMode(IdealState.RebalanceMode.FULL_AUTO);
+    admin.addResource(clusterName, testResource, idealState);
+
+    admin.enableResource(clusterName, testResource, true);
```

Comment:
Do you need to add a verifier check here? It might take some time for the controller to rebalance.

Reply:
TFTR. May I ask are you suggesting verify if the resource is enabled? In my understanding, if not, then we won't get the expected exception for nonexistent partitions in the next try/catch block.

Reply:
Please correct me if I am wrong. For example, I guess if we get "resource not added" exception while "nonexistent partition" is expected, we would know the resource is not correctly added.